upcloud/resource_upcloud_objectstorage.go

```diff
@@ -0,0 +1,205 @@
+package upcloud
+
+import (
+	"context"
+	"github.com/UpCloudLtd/upcloud-go-api/upcloud"
+	"github.com/UpCloudLtd/upcloud-go-api/upcloud/request"
+	"github.com/UpCloudLtd/upcloud-go-api/upcloud/service"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func resourceUpCloudObjectStorage() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceObjectStorageCreate,
+		ReadContext:   resourceObjectStorageRead,
+		UpdateContext: resourceObjectStorageUpdate,
+		DeleteContext: resourceObjectStorageDelete,
+		Schema: map[string]*schema.Schema{
+			"size": {
+				Description:  "The size of the object storage bucket in gigabytes",
+				Type:         schema.TypeInt,
+				Required:     true,
+				ValidateFunc: validation.IntInSlice([]int{250, 500, 1000}),
+			},
+			"access_key": {
+				Description:  "The access key used to identify user",
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringLenBetween(4, 255),
+			},
+			"secret_key": {
+				Description:  "The secret key used to authenticate user",
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringLenBetween(8, 255),
+			},
+			"zone": {
+				Description: "The zone in which the object storage bucket will be created",
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+			},
+			"name": {
+				Description: "The name of the object storage bucket to be created",
+				Optional:    true,
+				Type:        schema.TypeString,
+			},
+			"description": {
+				Description: "The description of the object storage bucket to be created",
+				Optional:    true,
+				Type:        schema.TypeString,
+				ValidateFunc: validation.StringLenBetween(1, 255),
+			},
+			"url": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"used_space": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"state": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"created": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceObjectStorageCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	var (
+	    diags diag.Diagnostics
+		req request.CreateObjectStorageRequest
+	)
+
+	client := m.(*service.Service)
+
+	req.Size = d.Get("size").(int)
+	req.Zone = d.Get("zone").(string)
+	req.AccessKey = d.Get("access_key").(string)
+	req.SecretKey = d.Get("secret_key").(string)
+
+	if v, ok := d.GetOk("name"); ok {
+		req.Name = v.(string)
+	}
```

Comment:
```req.Name = d.Get("name")``` 
(also for description)

essentially the same here, as will end up with zero val anyways